osu.Game/Utils/GeometryUtils.cs

```diff
@@ -218,5 +219,137 @@ private static IEnumerable<Vector2> enumerateStartAndEndPositions(IEnumerable<IH
 
                 return new[] { h.Position };
             });
+
+        #region welzl_helpers
+
+        // Function to check whether a point lies inside or on the boundaries of the circle
+        private static bool isInside((Vector2, float) c, Vector2 p)
```

Comment:
When using tuples like this, please name the members, rather than allowing the default inscrutable `Item{1,2}` crap

```suggestion
        private static bool isInside((Vector2 centre, float radius) c, Vector2 p)
```